integration_tests/test_search.py

```diff
@@ -323,34 +400,116 @@ class SpecificAuthor(FirstMatch):
-            return
+        subtitle = self._field('subtitle', first)
+        if self.accept_title and (self.accept_title in title or self.accept_title in subtitle):
+            return True
+
         # We have failed.
-        eq_(self.author, author)
+        if hasattr(expect, 'match'):
+            expect = expect.pattern
+        eq_(expect, first.contributors)
 
     def evaluate_hits(self, hits):
-        author = self.author
-        threshold = self.threshold
-        authors = [self._field('author', x) for x in hits]
-        author_matches = [x for x in authors if x == author]
-        self.assert_ratio(author_matches, authors, threshold)
+        last_role = None
+        last_title = None
+        author = self.original_kwargs['author']
+        authors = [hit.contributors for hit in hits]
+        author_matches = []
+        for hit in hits:
+            role = self.author_role(author, hit)
+            author_matches.append(role is not None)
+            last_role = role
+            last_title = hit.title
+        self.assert_ratio(author_matches, authors, self.threshold)
+
+
+class SpecificSeries(Common):
+    """Verify that results come from a certain series of books."""
+
+    def evaluate(self, results):
+        successes = []
+        diagnostics = []
+        for result in results:
+            success, should_have_matched = self.evaluate_one(result)
+            if success:
+                successes.append(result)
+            diagnostics.append(should_have_matched)
+        self.assert_ratio(successes, diagnostics, self.threshold)
+
+    def evaluate_one(self, result):
+        expect_author = self.kwargs.get('author')
+        expect_series = self.kwargs.get('series')
+
+        # Ideally a series match happens in the .series, but sometimes
+        # it happens in the .title.
+
+        actual_series = result.series or ""
+        series_match, details = self._match_scalar(
+            actual_series, expect_series, inclusive=True
+        )
+        actual_title = result.title
+        title_match, details = self._match_scalar(
+            actual_title, expect_series, inclusive=True
+        )
+
+        # Either way, if an author is specified, it means a book with
+        # a matching title by a different author is not part of the
+        # series.
+        if expect_author:
+            author_match, match, details = self._match_author(
+                expect_author, result
+            )
+        else:
+            author_match = True
+        actual = (actual_series, actual_title, result.author,
+                  result.sort_author, series_match, title_match, author_match)
+        return (series_match or title_match) and author_match, actual
 
 
 class SearchTest(object):
     """A test suite that runs searches and compares the actual results
     to some expected state.
     """
 
+    expected_failures = []
+    unexpected_successes = []
+
```

Comment:
Now we get into changes to specific tests -- mostly changing the way we evaluate tests -- and new tests.